### PR TITLE
feat: improve error handling of table

### LINKED ISSUE #147: Improve error handling of `Table`
### Is your feature request related to a problem?

In some cases, error handling is not ideal (see solution for details).

### Desired solution

* Validate inputs of a function as the first step
* Raise an appropriate exception with an appropriate message if issues are found
* Document in the docstring that the exception is raised and under which condition
* Test that the correct exception with the correct message is raised

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by robmeth**
Closes #147.

### Summary of Changes

feat: Added `WrongFileExtensionError`

* Validate inputs of a function as the first step
* Raise an appropriate exception with an appropriate message if issues are found
* Document in the docstring that the exception is raised and under which condition
* Test that the correct exception with the correct message is raised

Co-authored-by: Marsmaennchen221 <47296670+Marsmaennchen221@users.noreply.github.com>

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.2) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/5022765342)

| Descriptor  |                                 Linter                                 |Files|Fixed|                            Errors                             |Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|---------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.2/descriptors/python_black)          |   34|   14|                                                              0|2.0s        |
|✅ PYTHON    |[mypy](https://megalinter.io/6.22.2/descriptors/python_mypy)            |   34|     |                                                              0|2.37s       |
|❌ PYTHON    |[ruff](https://megalinter.io/6.22.2/descriptors/python_ruff)            |   34|   15|[20](https://github.com/Safe-DS/Stdlib/actions/runs/5022765342)|0.1s        |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.2/descriptors/repository_git_diff)|yes  |     |no                                                             |0.03s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/5022765342)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/data/tabular/containers/_table.py [hunk 1]
```diff
@@ -969,12 +999,14 @@ def slice_rows(
...
-            raise ValueError("The given index is out of bounds")
+        if end < start:
+            raise IndexOutOfBoundsError(slice(start, end))
+        if start < 0 or end < 0 or start > self.number_of_rows or end > self.number_of_rows:
+            raise IndexOutOfBoundsError(start if start < 0 or start > self.number_of_rows else end)
 
         new_df = self._data.iloc[start:end:step]
         new_df.columns = self._schema.column_names
-        return Table._from_pandas_dataframe(new_df)
+        return Table._from_pandas_dataframe(new_df, self._schema)
```
Comment (alex-senger) [inline src/safeds/data/tabular/containers/_table.py:1009]: Why would you keep the old schema on a new table?

Comment (Marsmaennchen221) [reply src/safeds/data/tabular/containers/_table.py:1009]:
```suggestion
        return Table._from_pandas_dataframe(new_df)
```

File: src/safeds/data/tabular/containers/_table.py [hunk 2]
```diff
@@ -328,6 +347,8 @@ def __eq__(self, other: Any) -> bool:
             return NotImplemented
         if self is other:
             return True
+        if self.number_of_rows == 0 and other.number_of_rows == 0:
+            return self._schema == other._schema
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_table.py:351]:
```suggestion
            return self.column_names == other.column_names
```

**Review by alex-senger — APPROVED** [review]
LGTM

**Pull request merged**